napari/_qt/qt_main_window.py

```diff
@@ -654,7 +654,10 @@ def add_plugin_dock_widget(
         full_name = plugin_menu_item_template.format(plugin_name, widget_name)
         if full_name in self._dock_widgets:
             dock_widget = self._dock_widgets[full_name]
-            dock_widget.show()
+            if dock_widget.isVisible():
+                dock_widget.hide()
+            else:
+                dock_widget.show()
```

Comment:
this was something I added when I was trying to get the behavior to work properly when a widget was undocked and then you toggled the widget from the menu.  Right now, it will call `add_plugin_dock_widget` when an action is toggled.  I suppose I should change that?